create_jenkins_job.py

```diff
@@ -363,7 +359,7 @@ def create_job(os_name, job_name, template_file, additional_dict):
         if os_name == 'windows':
             job_name = job_name[:15]
         test_args_default = os_configs.get(os_name, data).get('test_args_default', data['test_args_default'])
-        test_args_default = test_args_default.replace('--retest-until-pass', '--retest-until-fail') + " --ctest-args -LE linter --pytest-args -m 'not linter'"
+        test_args_default = test_args_default.replace('--retest-until-pass', '--retest-until-fail').replace('--ctest-args -LE xfail', "--ctest-args -LE '(linter|xfail)'").replace("--pytest-args -m 'not xfail'", "--pytest-args -m 'not linter and not xfail'")
```

Comment:
Although there's no code style guidelines the replacements here might be worth splitting into multiple lines for human readability.

Reply:
Done in fdc53f6b4855d7ba524cd202847f00356bcf12ef.